src/message-store-sql.ts

```diff
@@ -218,7 +287,50 @@ export class MessageStoreSql implements MessageStore {
     });
 
     const message = decodedBlock.value as GenericMessage;
+    // If encodedData is stored within the MessageStore we include it in the response.
+    // We store encodedData when the data is below a certain threshold.
+    // https://github.com/TBD54566975/dwn-sdk-js/pull/456
+    if (message !== undefined && encodedData !== undefined && encodedData !== null) {
+      (message as any).encodedData = encodedData;
+    }
     return message;
   }
 
+  /**
+   * Gets the pagination Message Cid if there are additional messages to paginate.
+   * Accepts more messages than the limit, as we query for additional records to check if we should paginate.
+   *
+   * @param messages a list of messages, potentially larger than the provided limit.
+   * @param limit the maximum number of messages to be returned
+   *
+   * @returns the pruned message results and an optional paginationMessageCid
+   */
+  private async getPaginationResults(
+    messages: Promise<GenericMessage>[], limit?: number
+  ): Promise<{ messages: GenericMessage[], paginationMessageCid?: string }>{
+    if (limit !== undefined && messages.length > limit) {
+      messages = messages.slice(0, limit);
+      const lastMessage = messages.at(-1);
+      return {
+        messages             : await Promise.all(messages),
+        paginationMessageCid : lastMessage ? await Message.getCid(await lastMessage) : undefined
+      };
+    }
+
+    return { messages: await Promise.all(messages) };
+  }
+
+  private getOrderBy(
+    messageSort?: MessageSort
+  ):{ property: 'dateCreated' | 'datePublished' | 'messageTimestamp', direction: SortOrder } {
+    if(messageSort?.dateCreated !== undefined)  {
+      return  { property: 'dateCreated', direction: messageSort.dateCreated };
+    } else if(messageSort?.datePublished !== undefined) {
+      return  { property: 'datePublished', direction: messageSort.datePublished };
+    } else if (messageSort?.messageTimestamp !== undefined) {
+      return  { property: 'messageTimestamp', direction: messageSort.messageTimestamp };
+    } else {
+      return  { property: 'messageTimestamp', direction: SortOrder.Ascending };
```

Comment:
I kinda expect the default to have a direction of descending, as usually as a consumer I'd be more interested with recent events rather than really old ones. 

Is there a particular reason you went with ascending here?

Reply:
We didn't really have an in-depth discussion about this when implementing it in `dwn-sdk-js` and `MessageStoreLevel`.

The `RecordsQuery` default sort was `dateCreated` in ascending order, so we kept the same order for the default of other records type, except using the default property `messageTimestamp`.

But good call out, might be good for a discussion to see what consumers of `web5-js` would expect, now that we have pagination.